src/colorizer/CanvasOverlay.ts

```diff
@@ -123,62 +134,107 @@ export default class CanvasWithOverlay extends ColorizeCanvas {
+  }
+
+  public getScaleInfo(): CanvasScaleInfo {
+    return this.innerCanvas.getScaleInfo();
+  }
+
   get domElement(): HTMLCanvasElement {
     // Override base ColorizeCanvas getter with the composited canvas.
-    return this.canvas;
+    return this.canvasElement;
+  }
+
+  setOnFrameLoadCallback(callback: (result: FrameLoadResult) => void): void {
+    this.onFrameLoadCallback = callback;
+    this.innerCanvas.setOnFrameLoadCallback(callback);
+  }
+
+  dispose(): void {
+    this.innerCanvas.dispose();
   }
 
   public setResolution(width: number, height: number): void {
-    this.canvasSize.x = width;
-    this.canvasSize.y = height;
-    super.setResolution(width, height);
+    this.innerCanvasSize.x = width;
+    this.innerCanvasSize.y = height;
+    this.innerCanvas.setResolution(width, height);
     this.render();
   }
 
   public getIdAtPixel(x: number, y: number): number {
     const headerHeight = this.headerSize.y;
-    return super.getIdAtPixel(x, y - headerHeight);
+    return this.innerCanvas.getIdAtPixel(x, y - headerHeight);
   }
 
   // Getters/Setters ////////////////////////////////
 
-  updateScaleBarStyle(style: Partial<ScaleBarStyle>): void {
+  public updateScaleBarStyle(style: Partial<ScaleBarStyle>): void {
     this.scaleBarStyle = { ...this.scaleBarStyle, ...style };
   }
 
-  updateTimestampStyle(style: Partial<TimestampStyle>): void {
+  public updateTimestampStyle(style: Partial<TimestampStyle>): void {
     this.timestampStyle = { ...this.timestampStyle, ...style };
   }
 
-  updateInsetBoxStyle(style: Partial<InsetBoxStyle>): void {
+  public updateInsetBoxStyle(style: Partial<InsetBoxStyle>): void {
     this.insetBoxStyle = { ...this.insetBoxStyle, ...style };
   }
 
-  updateLegendStyle(style: Partial<LegendStyle>): void {
+  public updateLegendStyle(style: Partial<LegendStyle>): void {
     this.legendStyle = { ...this.legendStyle, ...style };
   }
 
-  updateHeaderStyle(style: Partial<HeaderStyle>): void {
+  public updateHeaderStyle(style: Partial<HeaderStyle>): void {
     this.headerStyle = { ...this.headerStyle, ...style };
   }
 
-  updateFooterStyle(style: Partial<FooterStyle>): void {
+  public updateFooterStyle(style: Partial<FooterStyle>): void {
     this.footerStyle = { ...this.footerStyle, ...style };
   }
 
-  setIsExporting(isExporting: boolean): void {
+  // TODO: Move `isExporting` flag into state
+  public setIsExporting(isExporting: boolean): void {
     this.isExporting = isExporting;
   }
 
-  setCollection(collection: Collection | null): void {
-    this.collection = collection;
+  public setZoom(zoom: number): void {
+    this.zoomMultiplier = zoom;
+    if (this.innerCanvas instanceof ColorizeCanvas2D) {
+      this.innerCanvas.setZoom(zoom);
+    }
+    this.render();
+  }
+
+  public setPan(x: number, y: number): void {
+    this.panOffset.set(x, y);
+    if (this.innerCanvas instanceof ColorizeCanvas2D) {
+      this.innerCanvas.setPan(x, y);
+    }
+    this.render();
+  }
```

Comment:
TODO: Zoom and Pan are specific to 2D canvas! This interaction logic will definitely need some work later once we see how 3D panning/zooming is handled.